### PR TITLE
Generic damages

### PR DESCRIPTION
**Pull request by costashatz**
More damage types to the hexapod (DART):
- Complete leg removals
- ~~Simple body removals (not yet implemented)~~
- Blocked joints ~~(for now only in their zero configuration)~~
- Leg shortenings ~~(it is dirty at the moment)~~
- Free joints

~~**WIP**~~

### CODE REVIEW

**Comment by costashatz** [comment]
@dogoepp can you have a quick look at it?

I do not really like how I am handling the leg shortenings, but it was too complicated to do using DART (I will give it another try today).

**Comment by costashatz** [comment]
**Update:** @dogoepp  I implemented proper leg shortenings.

File: hexapod_dart/include/hexapod_dart/hexapod.hpp [hunk 1]
```diff
@@ -9,20 +9,29 @@
...
+    };
+
     class Hexapod {
     public:
         Hexapod() {}
 
-        Hexapod(std::string urdf_file, std::vector<int> broken_legs) : _skeleton(_load_urdf(urdf_file))
+        Hexapod(std::string urdf_file, std::vector<HexapodDamage> damages)
         {
+            _damages = damages;
```
Comment (dogoepp) [inline hexapod_dart/include/hexapod_dart/hexapod.hpp]: This seems redundant with the first lines of `_set_damages()`

Comment (costashatz) [inline hexapod_dart/include/hexapod_dart/hexapod.hpp]: True dat. Fixing it..

**Comment by dogoepp** [comment]
It seems that you solved your question. I've nothing to add :+1:

**Comment by costashatz** [comment]
> It seems that you solved your question. I've nothing to add

Thanks for checking the code though.. :smile:

**Comment by costashatz** [comment]
**Merging...**

**Pull request merged**